lib/spack/spack/modules.py

```diff
@@ -430,7 +433,7 @@ class Dotkit(EnvModule):
 
     prerequisite_format = None  # TODO : does something like prerequisite exist for dotkit?
 
-    default_naming_format = '{name}-{version}-{compiler.name}-{compiler.version}-{hash}'
+    default_naming_format = '{name}-{version}-{compiler.name}-{compiler.version}'
 
```

Comment:
I thought a bit if this could be done easily, but I see mainly two issues:

1. there should be a 1:1 map from spec to module file name to keep things simple and  consistent
2. variants cannot substitute an hash completely

The first point means that if we just leave out the hash or do something similar, like encoding in the name a subset of the variants, then it gets hard to maintain consistency when we ask for dependencies or prerequisites (as multiple specs will map to the same file name).

The second point instead means that the information in an an hash "strictly contains" the information of the variants (which make us fall into 1.) Also, having the variants substituting the hash may seem appealing for some packages... but I am sure that as soon as you will consider `boost` you will change your mind! :grin: